### PR TITLE
Workload Info Command 

### PR DESCRIPTION
**Pull request by jordan-rash**
```
└─❯ nex workload info rq6VMLUrDY64A6kfyuwyd1
Information about Workload rq6VMLUrDY64A6kfyuwyd1

           Name: counter
     Start Time: 2024-11-20 00:09:20
  Workload Type: direct_start
```

### CODE REVIEW

File: cmd/nex/workload.go [hunk 1]
```diff
@@ -208,6 +210,58 @@ func (s StopWorkload) Run(ctx context.Context, globals *Globals, w *Workload) er
...
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	w := columns.New("Information about Workload %s", i.WorkloadId)
+	w.AddRow("Name", resp.WorkloadSummary.Name)
+	w.AddRow("Start Time", resp.WorkloadSummary.StartTime)
+	w.AddRowIf("Run Time", resp.WorkloadSummary.Runtime, resp.WorkloadSummary.Runtime != "")
```
Comment (autodidaddict) [inline cmd/nex/workload.go:255]: haven't seen this `AddRowIf` before. Very nice.

File: node/internal/actors/control_api.go [hunk 2]
```diff
@@ -347,14 +347,12 @@ func (api *ControlAPI) handleAgentPing(m *nats.Msg) {
 	var workloadType, namespace, workloadID string
 
 	switch len(splitSub) {
-	case 4: // PREFIX.APING.<WORKLOAD_TYPE>.<NAMESPACE>
-		workloadType = splitSub[2]
-		namespace = splitSub[3]
+	case 4: // PREFIX.APING.<NAMESPACE>.<WORKLOAD_TYPE>
+		namespace = splitSub[2]
+		workloadType = splitSub[3]
```
Comment (autodidaddict) [inline node/internal/actors/control_api.go:352]: we should probably allow a keyword like `all` for the workload type to return all workloads within a given namespace. Either that or support 2 different token lengths: `prefix.aping.namespace` and `prefix.aping.namespace.all`

I think `prefix.aping.namespace` is more idiomatic NATS

File: node/internal/actors/direct_start_agent.go [hunk 3]
```diff
@@ -87,6 +87,28 @@ func (a *DirectStartAgent) Receive(ctx *goakt.ReceiveContext) {
 		ctx.Response(&actorproto.LameDuckResponse{
 			Success: true,
 		})
+	case *actorproto.PingWorkload:
+		a.logger.Debug("PingWorkload received", slog.String("name", ctx.Self().Name()), slog.String("workload", m.WorkloadId))
```
Comment (autodidaddict) [inline node/internal/actors/direct_start_agent.go:91]: we should probably refactor this into a function so the case statement arms don't get obscured.

**Review by autodidaddict — CHANGES_REQUESTED** [review]

File: cmd/nex/workload.go (hunk 1 above)
Comment (jordan-rash) [reply cmd/nex/workload.go:255]: Stole it from nats.go 🤣

File: node/internal/actors/control_api.go (hunk 2 above)
Comment (jordan-rash) [reply node/internal/actors/control_api.go:352]: ah yes, this is in a different branch I have locally..... essentially mapping `nex workloads ls` to a `$NEX.APING.NAMESPACE` request.  Will do that PR soon

File: node/internal/actors/direct_start_agent.go (hunk 3 above)
Comment (jordan-rash) [reply node/internal/actors/direct_start_agent.go:91]: I knew you were going to say this....

Comment (autodidaddict) [reply node/internal/actors/direct_start_agent.go:91]: Just call me `kevlint`

**Review by autodidaddict — APPROVED** [review]

**Review by autodidaddict — APPROVED** [review]

**Pull request merged**